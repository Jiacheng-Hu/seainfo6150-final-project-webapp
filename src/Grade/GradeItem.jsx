import React from "react";

const GradeItem = props => {
    
    return (
        <tr>
            <td>{props.gradeItem.id}</td>
            <td>{props.gradeItem.name}</td>
            <td>{props.gradeItem.credit}</td>
            <td>{props.gradeItem.year}</td>
            <td>{props.gradeItem.semester}</td>
            <td>{props.gradeItem.grade}</td>
        </tr>
    );
};

export default GradeItem;