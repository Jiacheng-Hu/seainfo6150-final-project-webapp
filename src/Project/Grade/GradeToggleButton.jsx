import React from "react";

const GradeToggleButton = props => {

    return (
        <button onClick={props.onClick} className="btn btn-default  btn-sm">{props.buttonText}</button>
    )
}

export default GradeToggleButton