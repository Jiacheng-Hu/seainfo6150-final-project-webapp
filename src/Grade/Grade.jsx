import React from "react";
import GradeItem from "./GradeItem";

const Grade = props => {
  
  let content;

  if (props.grade.length){
    content = (
      <table>
        <thead>
          <tr>
            <td>Course ID</td>
            <td>Course Name</td>
            <td>Credit Hour</td>
            <td>Associated Year</td>
            <td>Associated Semester</td>
            <td>Grade</td>
          </tr>
        </thead>
        <tbody>
          {props.grade.map((item) => (
            <GradeItem gradeItem = {item} key = {item.slug}/>
          ))}
        </tbody>
      </table>
    )
  }else{
    content = <div>You have no data!</div>
  }

  return props.grade.length ? content : null;

};

export default Grade;
