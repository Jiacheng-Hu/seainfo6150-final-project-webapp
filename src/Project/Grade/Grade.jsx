import React from "react";
import GradeItem from "./GradeItem";

const Grade = props => {
    let content;
    if(props.grade.length){
        content = (
            <div className="table-responsive container">
                <table className="table table-bordered  table-hover table-condensed">
                    <thead>
                        <tr>
                            <td>Course ID</td>
                            <td>Course Name</td>
                            <td>Credit Hour</td>
                            <td>Associated Year</td>
                            <td>Associated Semester</td>
                            <td>Grade</td>
                        </tr>
                    </thead>
                    <tbody>
                        {props.grade.map((item) => (
                            <GradeItem gradeItem = {item} key = {item.slug}/>
                        ))}
                    </tbody>
                </table>
            </div>
            
        )
    }else{
        content = null;
    }
    return <div>{content}</div>
}

export default Grade;