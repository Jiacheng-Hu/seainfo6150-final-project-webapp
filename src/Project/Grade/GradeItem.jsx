import React, {useState} from "react";
import GradeToggleButton from "./GradeToggleButton";

const GradeItem = props => {

    const[buttonText, setButtonText] = useState("Show More");

    function onClick(){
        if(buttonText === "Show More"){
            setButtonText("Show Less");
        }else{
            setButtonText("Show More");
        }
    }

    return (
        <tr>
            <td>{props.gradeItem.id}</td>
            <td>{props.gradeItem.name}</td>
            <td>{props.gradeItem.credit}</td>
            <td>{props.gradeItem.year}</td>
            <td>{props.gradeItem.semester}</td>
            <td>
                <div>total: {props.gradeItem.grade}</div>
                {
                    buttonText === "Show Less" ? 
                    <>
                    <div>attendance: {props.gradeItem.attendance}</div>
                    <div>assignment: {props.gradeItem.assignment}</div>
                    <div>exam: {props.gradeItem.exam}</div>
                    </> : null
                }
                <GradeToggleButton onClick={onClick} buttonText={buttonText}/>
            </td>
        </tr>
    )
}

export default GradeItem;