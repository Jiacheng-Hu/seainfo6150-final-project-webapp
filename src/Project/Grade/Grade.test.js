import React from "react";
import Grade from "./Grade";


const grade = [
    {
        slug: "170201-program-structure-and-algorithms",
        id: "170201",
        name: "Program Structure and Algorithms",
        credit: "4",
        year: "2017",
        semester: "Fall",
        grade: "A",
        attendance: "A",
        assignment: "A",
        exam: "A"
    },
    {
        slug: "170202-database-principle",
        id: "170202",
        name: "Database Principle",
        credit: "4",
        year: "2017",
        semester: "Fall",
        grade: "A-",
        attendance: "A",
        assignment: "A-",
        exam: "A-"
    }
];

describe("Grade tests", () => {
    it("renders correctly", () => {
        const {container} = render(<Grade grade={grade}/>);
        expect(container).toMatchSnapshot();
    });
});