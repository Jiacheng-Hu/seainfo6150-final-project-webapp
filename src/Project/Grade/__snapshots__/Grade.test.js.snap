// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grade tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="table-responsive container"
    >
      <table
        class="table table-bordered  table-hover table-condensed"
      >
        <thead>
          <tr>
            <td>
              Course ID
            </td>
            <td>
              Course Name
            </td>
            <td>
              Credit Hour
            </td>
            <td>
              Associated Year
            </td>
            <td>
              Associated Semester
            </td>
            <td>
              Grade
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              170201
            </td>
            <td>
              Program Structure and Algorithms
            </td>
            <td>
              4
            </td>
            <td>
              2017
            </td>
            <td>
              Fall
            </td>
            <td>
              <div>
                total: 
                A
              </div>
              <button
                class="btn btn-default  btn-sm"
              >
                Show More
              </button>
            </td>
          </tr>
          <tr>
            <td>
              170202
            </td>
            <td>
              Database Principle
            </td>
            <td>
              4
            </td>
            <td>
              2017
            </td>
            <td>
              Fall
            </td>
            <td>
              <div>
                total: 
                A-
              </div>
              <button
                class="btn btn-default  btn-sm"
              >
                Show More
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
